backend.go

```diff
@@ -0,0 +1,44 @@
+package gitobj
+
+import (
+	"io"
+
+	"github.com/git-lfs/gitobj/pack"
+	"github.com/git-lfs/gitobj/storage"
+)
+
+// NewFilesystemBackend initializes a new filesystem-based backend.
+func NewFilesystemBackend(root, tmp string) (storage.Backend, error) {
```

Comment:
Nit: what do you think about breaking the related constructors and types into their own files in package `storage`.

Reply:
I wanted to do that, but we can't, because the `newFileStorer` and `newMemoryStorer` functions are private.  I'll update the commit message to reflect that.